docs/concepts.md

```diff
@@ -0,0 +1,320 @@
+
+Time spent in the ReplicasLaunching stock is influential on system behaviour because
+Requests can continue to arrive while launching takes place. In the original designs
+this was substantially more detailed; for this generation of the design I chose to
+ignore the details while rebuilding the core simulator framework. A lot of improvements
+to simulation accuracy will probably come from breaking that Stock into finer detail. 
+
+Replicas are represented with the `ReplicaEntity`, a specialisation of Entity. The
+specialisation holds logic necessary to create and delete `Endpoints` in the Kubernetes
+API that the KPA consults to determine the current number of Replicas running.
+
+### Example: Requests
+
+Indirectly, Requests are signal that the Knative Pod Autoscaler is trying to respond
+to. In a traditional simulation these are called "arrivals". Unlike traditional
+simulation, the configuration of the simulated system varies throughout the simulation
+as the KPA changes its desired Replica count.
+
+The Movements graph for Requests is:
+ 
+```
+               +---------------------+
+               |                     |
+               V                     |
+TrafficSource -+-> RequestsBuffered -+-> RequestsProcessing --> RequestsComplete
+                                     |
+                                     +-> RequestsFailed
+```
+
+The diagram shows five possible Movements:
+
+* `arrive_at_buffer`, from TrafficSource to RequestsBuffered. These are scheduled
+  during the creation of the simulation
+* `send_to_replica`, from RequestsBuffered to RequestsProcessing
+* `complete_request`, from RequestsProcessing to RequestsComplete
+* `buffer_backoff`, from RequestsBuffered back into itself to simulate Activator behaviour
+* `exhausted_attempts`, from RequestsBuffered to RequestsFailed, representing a timeout
+
+In the happy path (`arrive_at_buffer`, `send_to_replica`, `complete_request`), this
+is a linear path. However, this is true only in aggregate: the exact path for each
+Request can vary. While all Requests will come from the TrafficSource and all of them
+will spend at least one Movement in RequestsBuffered, the precise RequestsProcessing /
+RequestsComplete stocks are not known when the original Request arrival is scheduled.
+There are multiple RequestsProcessing stocks, each belonging to a Replica.
+
+In the unhappy path, the RequestsBuffer recreates Activator logic by scheduling
+Movements back into itself in an exponential backoff pattern. On each such Movement it
+checks to see if any Replicas are in the ReplicasActive stock. If it finds any, it will
+pick one with a round-robin scheme and schedule a Movement of the Request to its
+RequestsProcessing. The RequestsProcessing stock will itself schedule a Movement
+into RequestsComplete. 
+
+This is probably the second major influence on Autoscaler behaviour. By Little's
+Law, a longer time to process a Request will mean that more Requests are being
```

Comment:
Maybe provide a link to Little's Law, such as https://en.wikipedia.org/wiki/Little%27s_law.